styx-api-service/src/test/java/com/spotify/styx/api/ResourceResourceTest.java

```diff
@@ -0,0 +1,257 @@
+        ImmutableMap.of("id", "resource2", "concurrency", 2)
+    ));
+  }
+
+  @Test
+  public void shouldGetResource() throws Exception {
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("GET", path("/resource1")));
+
+    assertThat(response, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(response, "id", equalTo("resource1"));
+    assertJson(response, "concurrency", equalTo(1));
+  }
+
+  @Test
+  public void shouldPostResource() throws Exception {
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("POST", path(""),
+            ByteString.encodeUtf8("{\"id\": \"resource2\", \"concurrency\": 2}")));
+
+    assertThat(response, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(response, "id", equalTo("resource2"));
+    assertJson(response, "concurrency", equalTo(2));
+
+    assertThat(storage.resource("resource2").get(), is(RESOURCE_2));
+  }
+
+  @Test
+  public void shouldUpdateResource() throws Exception {
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("PUT", path("/resource1"),
+            ByteString.encodeUtf8("{\"id\": \"resource1\", \"concurrency\": 21}")));
+
+    assertThat(response, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(response, "id", equalTo("resource1"));
+    assertJson(response, "concurrency", equalTo(21));
+
+    assertThat(storage.resource(RESOURCE_1.id()).get(), is(Resource.create(RESOURCE_1.id(), 21)));
+  }
+
+  @Test
+  public void shouldDeleteResource() throws Exception {
+    Response<ByteString> response =
+        awaitResponse(serviceHelper.request("DELETE", path("/resource1")));
+
+    assertThat(response, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+
+    assertThat(storage.resource(RESOURCE_1.id()).isPresent(), is(false));
+  }
+
+  @Test
+  public void shouldHandleMultipleResourcesIndependently() throws Exception {
+    // add resource2
+    awaitResponse(serviceHelper.request("POST", path(""),
+        ByteString.encodeUtf8("{\"id\": \"resource2\", \"concurrency\": 2}")));
+
+    // make sure both are listed
+    Response<ByteString> listResponse =
+        awaitResponse(serviceHelper.request("GET", path("")));
+    assertThat(listResponse, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(listResponse, "resources", hasSize(2));
+    assertJson(listResponse, "resources", containsInAnyOrder(
+        ImmutableMap.of("id", "resource1", "concurrency", 1),
+        ImmutableMap.of("id", "resource2", "concurrency", 2)
+    ));
+
+    // change resource2
+    Response<ByteString> putResponse =
+        awaitResponse(serviceHelper.request("PUT", path("/resource2"),
+            ByteString.encodeUtf8("{\"id\": \"resource2\", \"concurrency\": 3}")));
+    assertThat(putResponse, hasStatus(belongsToFamily(StatusType.Family.SUCCESSFUL)));
+    assertJson(putResponse, "id", equalTo("resource2"));
```

Comment:
Try not to assert things that are not essential for the test. Checking the put response object is already done in the `shouldUpdateResource` test case.